### PR TITLE
Build native desktop SDK in MSVC conformance mode (/permissive-)

### LINKED ISSUE #255: MSVC build logic should specify /permissive-
**Problem**
Due to originally being built before the standardization of C++ and sticking to those decisions for a bit too long, MSVC allows for non-standard behaviors. This creates problems by accidentally introducing non-standard code prior to a Pull Request.

**Describe the solution you'd like.**
Set /permissive- in all .vcxproj files. This way all MSVC code is built adhering to standard specified behavior.

### PR DESCRIPTION
**Pull request by bmehta001**
## What

Enable MSVC **conformance mode** (`/permissive-`) for the SDK's own native library builds, and fix the conformance errors this surfaces.

## Why

Per #255, MSVC's default (permissive) mode lets non-standard constructs slip into the codebase unnoticed. `/permissive-` makes the compiler reject them at build time.

## Scope and gating

`/permissive-` is applied via a dedicated `Solutions/conformance.props` (a single `ConformanceMode=true` `ItemDefinitionGroup`) that is imported **explicitly, per-project**, by the five native SDK library projects only:

- `win32-lib`, `win32-dll`, `win32-mini-lib`, `win32-mini-dll`, `net40`

It is imported right after each project's `build.props` import, so the setting wins ItemDefinitionGroup ordering.

It is deliberately **not** put on the shared `Solutions/build.props`, because `build.props` is also imported by vendored `sqlite`/`zlib`, the test projects (gtest/gmock/UnitTests/FuncTests), and the samples — none of which we want to force into conformance mode. The UWP `win10-*` SDK projects also do not import it, because `/permissive-` is incompatible with C++/CX (`/ZW`); their shared SDK sources are still conformance-checked through the `win32` library builds.

## Conformance fixes (C++/CLI net40)

Enabling `/permissive-` on the managed `net40` project surfaced two two-phase-lookup errors in shared CX code, both fixed here:

| File | Error | Fix |
|---|---|---|
| `lib/shared/EventPropertiesCX.cpp` | `C3878`/`C2065` — `map<string,T>::iterator` is a dependent type | add `typename` |
| `lib/shared/PlatformHelpers.h` | `C3861` — `FromPlatformString` used by `FromPlatformMap` templates before its declaration; dependent call fell to ADL (`Platform::`) | forward-declare it before the templates |

## Validation (VS 2026, PlatformToolset v145, x64 Release)

- `/permissive-` confirmed present on `win32-lib`'s cl invocations; clean **Rebuild** of all five SDK library projects → **0 conformance errors**.
- `net40` (C++/CLI) → `Microsoft.Applications.Telemetry.Windows.NET.dll`.
- Confirmed the scoping: a standalone `sqlite:Rebuild` shows **0** `/permissive-` occurrences (vendored deps and tests are unaffected).

> Note: `net40` is .NET Framework 4.0 C++/CLI. Building it locally requires the v4.0 reference-assembly targeting pack (the SDK CI provides it); a machine missing it will hit `C1107: could not find assembly 'mscorlib.dll'`, an environment prerequisite unrelated to this change.

Resolves #255.

### CODE REVIEW

File: Solutions/build.props [hunk 1]
```diff
@@ -13,4 +13,16 @@
...
+    non-standard constructs are caught at build time. UWP / Windows Store
+    (AppContainerApplication) projects are excluded because they compile as
+    C++/CX (/ZW), which MSVC rejects in combination with /permissive-.
+    See issue #255.
+  -->
+  <ItemDefinitionGroup Condition="'$(AppContainerApplication)' != 'true'">
+    <ClCompile>
+      <ConformanceMode>true</ConformanceMode>
+    </ClCompile>
+  </ItemDefinitionGroup>
```
[the remark this answers is not in the document]
Comment (bmehta001) [reply Solutions/build.props]: Good catch — you're right, and my PR description was wrong. `Solutions/build.props` is imported well beyond the SDK library projects; `git grep -l build.props -- '*.vcxproj'` confirms `sqlite/sqlite.vcxproj`, `third_party/Solutions/zlib/vc14/zlibvc.vcxproj`, and `tests/functests/FuncTests.vcxproj` / `tests/unittests/UnitTests.vcxproj` (plus gtest/gmock and samples) all import it. So the gated `ItemDefinitionGroup` was forcing `/permissive-` onto vendored deps and the unvalidated test projects too.

Fixed in 1afe1efd: the `ConformanceMode` block is moved out of `build.props` into a dedicated `Solutions/conformance.props` that is imported **explicitly only** by the five native SDK library projects (`win32-lib`, `win32-dll`, `win32-mini-lib`, `win32-mini-dll`, `net40`), right after their `build.props` import so it still wins ItemDefinitionGroup ordering. `build.props` is reverted.

Verified locally (VS 2026, v145, x64 Release): `/permissive-` is present on `win32-lib`'s cl invocations with 0 conformance errors, while a standalone `sqlite:Rebuild` now shows **0** `/permissive-` occurrences. Fixed.

**Review by baijumeswani — APPROVED** [review]
